Coding/server/lib/User/User.cpp

```diff
@@ -0,0 +1,301 @@
+/*
+BSD 3-Clause License
+
+Copyright (c) 2021, NewTec GmbH
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice, this
+   list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+   contributors may be used to endorse or promote products derived from
+   this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+ * @file User.cpp
+ * @author Luis Moser
+ * @brief User class
+ * @date 07/20/2021
+ *
+ * @{
+ */
+
+#include <User.h>
+
+User* User::m_registeredUsers[MAX_REGISTERED_USERS];
+uint8_t User::m_numberOfRegisteredUsers = 0;
+CryptoServices User::m_crypto;
+
+User::User() :
+    m_username(),
+    m_hashedPassword(),
+    m_passwordSalt(),
+    m_permissions{},
+    m_numberOfPermissions(0)
+{
+}
+
+User::~User()
+{
+}
+
+int8_t User::getUserIdx(const String& username)
+{
+    int8_t retIdx = -1;
+
+    for (uint8_t userIdx = 0; userIdx < MAX_REGISTERED_USERS; userIdx++)
+    {
+        if ((nullptr != m_registeredUsers[userIdx])
+            && m_registeredUsers[userIdx]->m_username == username)
+        {
+            retIdx = userIdx;
+            break;
+        }
+    }
+    return retIdx;
+}
+
+bool User::checkAdminAccount()
+{
+    const String defaultUsername = "admin";
+    const String defaultPassword = "21091986";
```

Comment:
Don't hide them somewhere in a method. Move them at least to the class declaration scope.

Reply:
Fixed with f583666